src/include/miopen/convolution.hpp

```diff
@@ -99,9 +109,18 @@ struct ConvolutionAttribute
     /// * 1: Enabled/No.
     void Set(miopenConvolutionAttrib_t attr, int value);
     int Get(miopenConvolutionAttrib_t attr) const;
+
+    template <class Stream, std::enable_if_t<IsBinarySerializationRelated<Stream>{}, bool> = true>
+    friend Stream& operator<<(Stream& stream, ConvolutionAttribute& conv)
+    {
+        stream << conv.gfx90aFp16alt;
+        return stream;
+    }
 };
 
-struct ConvolutionDescriptor : miopenConvolutionDescriptor
+struct ConvolutionDescriptor
+    : miopenConvolutionDescriptor,
+      OperatorDescriptorImpl<ConvolutionDescriptor, solver::Primitive::Convolution>
```

Comment:
This should use the `Serializable` class, with a `Visit` method, like how did in #1290: https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1290/files#diff-082c9699504013c0a45dd886dccf551d0b11f16b16677c597deff1471ee5e84eR138

Reply:
Maybe we dont want to make all our descriptors inherit from `Serializable` as the `Serialize` methods are in the headers, so it could impact compile-time.

Just having a `Visit` method should be enough to implement serialization in a single .cpp file.

Reply:
Didn't notice we have msgpack as dependency.

Reply:
Hmmm, not yet. I c

Reply:
@junliume would you mind having nlohmann/json as a dependency? If so, what way would you prefer: copying the header or externally referencing?